src/modules/cache-client/cache-client.service.ts

```diff
@@ -681,18 +667,48 @@ export class CacheClient implements ICacheClient {
     );
   }
 
-  private async refreshToken(): Promise<AuthTokens | undefined> {
-    if (!this.refresh_token) return undefined;
+  private async refreshToken(): Promise<void> {
+    if (!this.isBrowser && !this.refresh_token) return undefined;
     getLogger().info('[CACHE CLIENT] refreshing tokens');
-    const { data } = await this._httpClient.get<{
-      token: string;
-      refreshToken: string;
-    }>(
+    const res = await this._httpClient.get<AuthTokens>(
       `/refresh_token${
         this.isBrowser ? '' : `?refresh_token=${this.refresh_token}`
       }`
     );
     getLogger().debug('[CACHE CLIENT] refreshed tokens fetched');
-    return data;
+    if (!this.isBrowser) {
+      this.setTokens(res);
+    }
+  }
+
+  /**
+   * Saves access and refresh tokens from login response
+   *
+   * @param res Response from login request
+   */
+  private setTokens({ headers, data }: AxiosResponse) {
+    let token: AuthTokens['token'] | undefined;
+    let refreshToken: AuthTokens['refreshToken'] | undefined;
+    if (headers['set-cookie']) {
+      const cookies = setCookie.parse(headers['set-cookie'], {
+        decodeValues: false,
+        map: true,
+      });
+      const tokenCookie = cookies['token'];
+      const refreshTokenCookie = cookies['refreshToken'];
+
+      if (tokenCookie && refreshTokenCookie) {
+        token = tokenCookie.value;
+        refreshToken = refreshTokenCookie.value;
+      }
+    }
+    if (!token || !refreshToken) {
+      token = data.token;
+      refreshToken = data.refreshToken;
+    }
```

Comment:
What's the rationale for these lines? It seems to be covering the case where the tokens aren't returned in the cookie headers. Is this right? I guess this could be useful to have but just trying to understand a bit better 😄

Reply:
I left this temporarily until we will change ssi-hub to send tokens only in cookies